Add documentation for new my-PV integration (#45204)
home-assistant/home-assistant.github.io: source/_integrations/my_pv.markdown

Password:
  description: "Password or device key of your my-PV device."
{% endconfiguration_basic %}

Older firmware versions of the my-PV hardware do not require a password, this will be added by upcoming firmware updates. When no custom password is set you have to use the **devicekey** which can be found under the ⓘ info menu of your my-PV device. For the HEA•THOR IoT you can find the **devicekey** on the device label.

You can update the password through the web interface of your my-PV device.

### Login to my-PV (Cloud API)

{% configuration_basic %}
Serial number:
  description: "The 16-digit serial number of your my-PV device. You can find it on the device label."
Token:
  description: "The my-PV Cloud API access token."
{% endconfiguration_basic %}

To generate an API token go to your [my-PV Cloud](https://live.my-pv.com/) and navigate to ⚙️ > **Device Setup**. Enable **Advanced Settings**, navigate to **API** and **Generate API Access Token**. Use this when setting up the my-PV integration.

## Data updates

The **my-PV** integration {% term polling polls %} data from the Local API every 5 seconds and my-PV Cloud every 30 seconds to ensure timely updates.

> 

Reviewer: ```suggestion

## Supported functionality

The **my-PV** integration provides the following entity.

### Water heater

- **Description**: Controls your my-PV device and allows you to monitor and adjust water heating.
- **Features**: Turn the device on/off, set target temperature, monitor current temperature.
- **Available for devices**: All supported my-PV devices with heating capability.

```
we usually add a section about the supported functionality. https://github.com/home-assistant/home-assistant.io/blob/current/source/_integrations/_integration_docs_template.markdown?plain=1#L91, https://developers.home-assistant.io/docs/documenting/integration-docs-examples#basic-structure-of-an-integration-page

Author: I know, but that's only a hard requirement for gold tier integrations. There are more entities and functionality to come once the first HA Core pull request is accepted. Documentation will be improved once that's done.

---
title: my-PV
description: Instructions on how to integrate my-PV devices into Home Assistant.
ha_category:
  - Binary sensor

>   - Sensor

Reviewer: ```suggestion
```

  - Water heater
ha_release: 2026.6
ha_iot_class: Local Polling
ha_codeowners:
  - '@my-PV'
  - '@rrooggiieerr'
ha_domain: my_pv
ha_platforms:
  - binary_sensor
  - sensor

>   - number

Reviewer: ```suggestion
```

  - water_heater
ha_integration_type: integration
ha_config_flow: true
ha_dhcp: true
ha_zeroconf: true
ha_quality_scale: bronze
---

The **my-PV** {% term integration %} is used to integrate with the devices of [my-PV](https://www.my-pv.com/). my-PV produces functional and innovative solutions for housing technology powered by solar electricity.

## Supported devices

The following devices are supported by this integration:
- AC ELWA 2
- AC•THOR range
- HEA•THOR IoT
- SOL•THOR

## Unsupported devices

The following devices are not supported by the integration:
- ELWA immersion heater
- WiFi Meter

> 

Reviewer: ```suggestion
```

The my-PV integration supports both the local API and the my-PV cloud API.

{% include integrations/config_flow.md %}

### Login to my-PV (Local API)

> Using the local API is the preferred way to connect to your my-PV device.
> 
> {% configuration_basic %}
> Password:
>   description: "Password or devicekey of your my-PV device."
> {% endconfiguration_basic %}

Reviewer: ```suggestion
{% configuration_basic %}
Host:
  description: "The IP address of your my-PV device. You can find it in your router or in the device's web interface."
Password:
  description: "Password or device key of your my-PV device."
{% endconfiguration_basic %}
```
Would I need to enter the host address during setup?

Older firmware versions of the my-PV hardware do not require a password, this will be added by upcoming firmware updates. When no custom password is set you have to use the **devicekey** which can be found under the ⓘ info menu of your my-PV device. For the HEA•THOR IoT you can find the **devicekey** on the device label.

You can update the password through the web interface of your my-PV device.

### Login to my-PV (Cloud API)

{% configuration_basic %}
Serial number:

>   description: "The serial number of your my-PV device."

Reviewer: ```suggestion
  description: "The 16-digit serial number of your my-PV device. You can find it on the device label."
```
If possible, I would be more specific here. Added a suggestion to illustrate what I mean. Please fix it so that it is correct for my-PV.

Author: It actually is 16 digits and printed on the device label

Token:

>   description: "The API token"

Reviewer: ```suggestion
  description: "The my-PV Cloud API access token."
```

---
title: my-PV
description: Instructions on how to integrate my-PV devices into Home Assistant.
ha_category:
  - Binary sensor
  - Sensor
  - Water heater
ha_release: 2026.6
ha_iot_class: Local Polling
ha_codeowners:
  - '@my-PV'
  - '@rrooggiieerr'
ha_domain: my_pv
ha_platforms:
  - binary_sensor
  - sensor
  - number
  - water_heater
ha_integration_type: integration
ha_config_flow: true
ha_dhcp: true
ha_zeroconf: true
ha_quality_scale: bronze
---

The **my-PV** {% term integration %} is used to integrate with the devices of [my-PV](https://www.my-pv.com/). my-PV produces functional and innovative solutions for housing technology powered by solar electricity.

## Supported devices

The following devices are supported by this integration:
- AC ELWA 2
- AC•THOR range
- HEA•THOR IoT
- SOL•THOR

## Unsupported devices

The following devices are not supported by the integration:
- ELWA immersion heater
- WiFi Meter

> {% endimportant %}

Reviewer: ```suggestion
```

Reviewer: I think this should fix the build
